FunctionalTableData/CellStyle.swift

```diff
@@ -160,10 +185,17 @@ extension CellStyle: Equatable {
 		equality = equality && lhs.accessoryType == rhs.accessoryType
 		equality = equality && lhs.selectionColor == rhs.selectionColor
 		equality = equality && lhs.backgroundColor == rhs.backgroundColor
-		equality = equality && lhs.backgroundView == rhs.backgroundView
```

Comment:
I think we need to expand this change a bit.
backgroundViewProvider should be used as part of the equality check (I think) because you can have the same cell key and state but provide a different backgroundViewProvider which should force it to refresh that cell.

I'm not sure how we make BackgroundViewProvider equatable though…

Reply:
give it an ID? but how not to clash 🤔

Reply:
We could force some `state:Equatable` that the user implements. 
The other problem with this approach is the background view is still associated to the ViewState instead of the cell and this makes reuse impossible. If we're scrolling through a long list and all cells have a backgroundView providers, those views will be initialized on the fly and not reused even if the cell types are the same.